File: solution_find_smallest.cpp
// C++ Solution - Find the smallest number in an array
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int num;
    int minNum = INT_MAX;
    
    // Read all integers from input until EOF
    while (cin >> num) {
        minNum = min(minNum, num);
    }
    
    // Output the minimum
    cout << minNum << endl;
    
    return 0;
}

File: solution_find_smallest.py
import sys

def findTheSmallestNumberInAnArray(input_str):
    """Find the smallest number in an array"""
    # Convert the input string to list of integers
    arr = list(map(int, input_str.split()))
    
    # Return the smallest element
    return str(min(arr))

def solve():
    lines = sys.stdin.readlines()
    if not lines:
        return
    
    # First line contains the array elements
    res = findTheSmallestNumberInAnArray(lines[0].strip())
    print(res)

if __name__ == "__main__":
    solve()
